metafields.go

```diff
@@ -0,0 +1,196 @@
+package shopify
+
+import (
+	"strconv"
+	"time"
+)
+
+// Metafields are a flexible way to attach additional information to a Shopify resource (e.g. Product, Collection, etc.).
+type Metafields []Metafield
+
+// GetByKey gets a metafield via its key
+func (m Metafields) GetByKey(key string) (interface{}, error) {
```

Comment:
If we are using Go 1.8+

```suggestion
func (m Metafields) GetByKey(key string) (any, error) {
```

Reply:
Unfortunately not. I think what we could do is next time we bump the actual Shopify API version up we obviously make a breaking change. Then during that breaking change, we could bring in the go version also. There is also this concept of V2, V3 which we have not done before https://go.dev/blog/v2-go-modules.